geode-management/src/main/java/org/apache/geode/management/internal/ClientClusterManagementService.java

```diff
@@ -54,34 +73,91 @@ private ClientClusterManagementService() {
     restTemplate.setErrorHandler(DEFAULT_ERROR_HANDLER);
   }
 
-  public ClientClusterManagementService(ClientHttpRequestFactory requestFactory) {
-    this();
-    this.restTemplate.setRequestFactory(requestFactory);
+  public ClientClusterManagementService(String host, int port, boolean useSSL, String username,
+      String password) {
+    this(host, port, useSSL ? createTrustAllSSLContext() : null,
+        useSSL ? new NoopHostnameVerifier() : null, username, password);
   }
 
-  public ClientClusterManagementService(String clusterUrl) {
+  public ClientClusterManagementService(String host, int port, SSLContext sslContext,
+      HostnameVerifier hostnameVerifier, String username, String password) {
     this();
+
     DefaultUriTemplateHandler templateHandler = new DefaultUriTemplateHandler();
-    templateHandler.setBaseUrl(clusterUrl);
+    String schema = (sslContext == null) ? "http" : "https";
+    templateHandler.setBaseUrl(schema + "://" + host + ":" + port + "/geode-management/v2");
     restTemplate.setUriTemplateHandler(templateHandler);
+
+    // HttpComponentsClientHttpRequestFactory allows use to preconfigure httpClient for
+    // authentication and ssl context
+    HttpComponentsClientHttpRequestFactory requestFactory =
+        new HttpComponentsClientHttpRequestFactory();
+
+    HttpClientBuilder clientBuilder = HttpClientBuilder.create();
+    // configures the clientBuilder
+    if (username != null) {
+      CredentialsProvider credsProvider = new BasicCredentialsProvider();
+      credsProvider.setCredentials(new AuthScope(host, port),
+          new UsernamePasswordCredentials(username, password));
+      clientBuilder.setDefaultCredentialsProvider(credsProvider);
+    }
+
+    clientBuilder.setSSLContext(sslContext);
+    clientBuilder.setSSLHostnameVerifier(hostnameVerifier);
+
+    requestFactory.setHttpClient(clientBuilder.build());
+    restTemplate.setRequestFactory(requestFactory);
+  }
+
+  public ClientClusterManagementService(ClientHttpRequestFactory requestFactory) {
+    this();
+    this.restTemplate.setRequestFactory(requestFactory);
+  }
+
+  public static SSLContext createTrustAllSSLContext() {
+    try {
+      SSLContext sslContext = SSLContext.getInstance("TLS");
```

Comment:
This should probably be using `SSLContext.getDefault()`. That way it gets initialized automatically and will also honor the various `javax.net.ssl.*` properties if set. Please make sure there is a pure client test that sets and uses these properties.